person.rb

```diff
@@ -7,17 +13,52 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    super
+  end
+
+  def correct_name
+    name
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
   end
 
   private
 
   def of_age?
     @age >= 18
   end
+end
 
-  public
+class Decorator < Nameable
+  attr_accessor :name
 
-  def can_use_services?
-    of_age? || @parent_permission
+  def initialize(nameable = 'Unknown')
+    super
```

Comment:
- [ ] Please accept my suggestion!


```suggestion
```

Reply:
@Rustamxon7 what's your suggestion about it